src/tokenizer.rs

```diff
@@ -1235,6 +1245,11 @@ impl<'a> Tokenizer<'a> {
                 '\\' => {
                     // consume
                     chars.next();
+
+                    if allow_escape {
```

Comment:
This behavior seems different than how MySqlDialect behaves

Specifically, with MySQL the escape characters are transformed into their literal values (`e.g. `'a\"b'` would be parsed to `a"b` while this PR would parse it to `a\"b`

What do you think about making this consistent?

Reply:
Ah yes I'll take a closer look at this to keep the same behavior for Mysql

Reply:
@alamb Looking into updating this and turns out I didn't follow the comment entirely and was unable to infer the inconsistency for mysql - could you clarify the problem once more? It seems Github/markdown unfortunately reformatted the expected and desired output in your example so that they became identical :sweat_smile:

Reply:
As in I think the tokenizer for mysql (assuming `self.unescape` is true) actually does the unescaping -- so a string with an escape character (`"\x20"`) would actually be tokenized as a space (`" "`)

Reply:
That makes sense! I've updated this have the new logic respect the unescape similar to mysql - let me know if that's what you had in mind!